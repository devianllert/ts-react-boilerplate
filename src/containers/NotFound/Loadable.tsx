import loadable, { ImportComponent } from '../../utils/loadable';

const LoadableNotFound = loadable((): ImportComponent => import(/* webpackChunkName: "notFound" */ './NotFound'), {
  fallback: null,
});

export default LoadableNotFound;
