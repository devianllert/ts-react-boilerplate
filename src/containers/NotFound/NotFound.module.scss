@import '../../design/design-system';

.container {
  display: flex;
  flex-direction: column;
  flex: 1;

  align-items: center;
  justify-content: center;

  min-height: 100vh;

  padding: $space-24;
}

.logo {
  margin: 0;

  font-size: $font-size-h1 * 2;
  font-weight: 200;

  line-height: $line-height-h1 * 2;
}

.title {
  margin: 0;

  font-size: $font-size-h2;

  line-height: $line-height-h2;
}

.subtitle {
  margin: $space-16 0;

  font-size: $font-size-h4;
  font-weight: 200;

  text-align: center;

  line-height: $line-height-h4;
}