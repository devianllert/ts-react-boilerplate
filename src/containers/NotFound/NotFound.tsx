import React, { ReactElement } from 'react';

const NotFound = (): ReactElement => (
  <div className="container">
    <h1>Page Not Found!</h1>
  </div>
);

export default NotFound;
