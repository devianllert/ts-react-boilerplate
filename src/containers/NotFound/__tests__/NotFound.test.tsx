import React from 'react';
import { cleanup, render } from '@testing-library/react';

import NotFound from '../NotFound';

import '@testing-library/jest-dom/extend-expect';

describe('<NotFound />', (): void => {
  afterEach(cleanup);

  it('should render and match the snapshot', (): void => {
    const { container } = render(<NotFound />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with initial state', (): void => {
    const { getByText } = render(<NotFound />);

    expect(getByText(/Page Not Found!/)).toBeDefined();
  });
});
