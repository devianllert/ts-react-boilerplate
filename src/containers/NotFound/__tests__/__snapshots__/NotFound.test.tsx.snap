// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render and match the snapshot 1`] = `
<div
  class="content"
>
  <h1
    class="logo"
  >
    404
  </h1>
  <h2
    class="title"
  >
    Page Not Found!
  </h2>
  <span
    class="subtitle"
  >
    Oops, looks like that page doesn\`t exists
  </span>
  <a
    class="button"
    href="/"
  >
    Go Home
  </a>
</div>
`;
